cmd/crypto/header.go

```diff
@@ -111,13 +113,13 @@ func (ssec) IsRequested(h http.Header) bool {
 // at least one SSE-C copy header. Regular SSE-C headers
 // are ignored.
 func (ssecCopy) IsRequested(h http.Header) bool {
-	if _, ok := h[SSECopyAlgorithm]; ok {
+	if h.Get(SSECopyAlgorithm) != "" {
```

Comment:
Not correct. The header can be set but be empty.

Reply:
h.Get() does a case insensitive comparison whereas the lookup style does not - this is causing test failures because the header defiinition is in lower case for 	SSECopyKeyMD5 = "X-Amz-Copy-Source-Server-Side-Encryption-Customer-Key-Md5". and the SSEKeyMD5. Since client can specify request in any case and the Http header interpretation should be agnostic to that, the lookup style needs to convert case before comparing

Reply:
Okay, good catch. Hm, I don't see how to solve that completely without ranging over all header keys - which is not probably not acceptable. Using the following approach:
```
_, ok := h[SSEHeader]
return ok || h.Get(SSEHeader) != ""
```
That detects a well-formed but empty entry(`X-Amz-Server-Side...`) and all other cases except a non-well-formed entry which is empty. `h.Get(SSEHeader) != ""` is IMO not enough since if I sent a request where `X-Amz-Server-Side-Encryption: ""` the server happily stores the object (in plaintext) whithout complaining. As additional step we can have a central point at the server were we copy and canonicalize the header keys. (not sure whether this is already done)...
Am I missing something here?

Reply:
@aead, tested further - your original lookup is correct and sufficient. The test(s)  in question itself was expecting incorrect result and previous implementation of hasSSE(Copy)CustomerHeader was incorrect in testing for non empty header value instead of checking for header presence. Will be fixing the tests in question and removing any changes I made to crypto/header.go

Reply:
:+1: